src/main/java/com/mcsunnyside/coreprotecttnt/Main.java

```diff
@@ -43,220 +52,243 @@ public void onEnable() {
+        ConfigurationSection section = Util.bakeConfigSection(getConfig(), "hanging");
+        if (!section.getBoolean("enable", true))
+            return;
+        if (e.getCause() == HangingBreakEvent.RemoveCause.PHYSICS || e.getCause() == HangingBreakEvent.RemoveCause.DEFAULT)
+            return; // We can't track them tho.
+
+        Block hangingPosBlock = e.getEntity().getLocation().getBlock();
+        String reason = probablyCache.getIfPresent(hangingPosBlock.getLocation());
+        if (reason != null) {
+            api.logRemoval("#" + e.getCause().name() + "-" + reason, hangingPosBlock.getLocation(), Material.matchMaterial(e.getEntity().getType().name()), null);
+        }
+    }
+
+    // EndCrystal rigged by entity (listener)
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
+    public void onEndCrystalHit(EntityDamageByEntityEvent e) {
+        if (!(e.getEntity() instanceof EnderCrystal))
+            return;
+        if (e.getDamager() instanceof Player) {
+            probablyCache.put(e.getEntity(), e.getDamager().getName());
+        } else {
+            if (probablyCache.getIfPresent(e.getDamager()) != null) {
+                probablyCache.put(e.getEntity(), probablyCache.getIfPresent(e.getDamager()));
+            } else if (e.getDamager() instanceof Projectile) {
+                Projectile projectile = (Projectile) e.getDamager();
+                if (projectile.getShooter() != null && projectile.getShooter() instanceof Player) {
+                    probablyCache.put(e.getEntity(), ((Player) projectile.getShooter()).getName());
                 }
             }
-            Location blockCorner = tnt.getLocation().clone().subtract(0.5, 0, 0.5);
-            for (Map.Entry<Location, String> entry : ignitedBlocks.entrySet()) {
-                if (entry.getKey().getWorld().equals(blockCorner.getWorld()) && entry.getKey().distance(blockCorner) < 0.5) {
-                    explosionSources.put(tnt, entry.getValue());
-                    break;
-                }
+        }
+    }
+
+    // Haning hit by entity (logger)
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
+    public void onHangingHit(EntityDamageByEntityEvent e) {
+        if (!(e.getEntity() instanceof Hanging))
+            return;
+        ConfigurationSection section = Util.bakeConfigSection(getConfig(), "itemframe");
+        if (!section.getBoolean("enable", true))
+            return;
+        ItemFrame itemFrame = (ItemFrame) e.getEntity();
+        if (itemFrame.getItem().getType().isAir() || itemFrame.isInvulnerable())
+            return;
+        if (e.getDamager() instanceof Player) {
+            probablyCache.put(e.getEntity(), e.getDamager().getName());
+            api.logInteraction(e.getDamager().getName(), itemFrame.getLocation());
+            api.logRemoval(e.getDamager().getName(), itemFrame.getLocation(), itemFrame.getItem().getType(), null);
```

Comment:
Are you sure it also should be logged as an interaction? CoreProtect only logs it as a removal: https://github.com/PlayPro/CoreProtect/blob/d9f5411e874ae0f1d7c618899bc2d3acad548223/src/main/java/net/coreprotect/listener/entity/HangingBreakByEntityListener.java#L115

Reply:
removed